damnit/ctxsupport/ctxrunner.py

```diff
@@ -304,9 +315,14 @@ def execute(self, run_data, run_number, proposal, input_vars) -> 'Results':
                     # Dependency within the context file
                     if annotation.startswith("var#"):
                         dep_name = annotation.removeprefix("var#")
-                        if dep_name in dep_results:
+                        match = fnmatch.filter(dep_results, dep_name)
+
+                        if len(match) == 1 and match[0] == dep_name:
                             dep_data = dep_results[dep_name]
                             kwargs[arg_name] = dep_data
+                        elif len(match) > 1:
+                            dep_data = {name: dep_results[name] for name in match}
+                            kwargs[arg_name] = dep_data
```

Comment:
While we're at it we can make this code slightly shorter:

```suggestion
                            kwargs[arg_name] = dep_results[dep_name]
                        elif len(match) > 1:
                            kwargs[arg_name] = {name: dep_results[name] for name in match}
```